src/yieldOptions/providers/protocols/stargate.ts

```diff
@@ -32,11 +36,16 @@ export class Stargate implements IProtocol {
       abi: sTokenAbi,
       functionName: "token"
     });
-    const pools = (await axios.get("https://yields.llama.fi/pools")).data;
 
-    const filteredPools: Pool[] = pools.data.filter((pool: Pool) => pool.chain === networkNames[chainId] && pool.project === "stargate")
-    const pool = filteredPools.find(pool => getAddress(pool.underlyingTokens[0]) === getAddress(token))
+    let pools = this.cache.get("llama-pools") as Pool[];
+    if (!pools) {
+      const res = (await axios.get("https://yields.llama.fi/pools")).data;
+      pools = res.data;
+      this.cache.set("llama-pools", pools);
+    }
 
+    const filteredPools: Pool[] = pools.filter((pool: Pool) => pool.chain === networkNames[chainId] && pool.project === "stargate")
+    const pool = filteredPools.find(pool => getAddress(pool.underlyingTokens[0]) === getAddress(token))
```

Comment:
we should filter by project before caching it. Otherwise we cache tons of data that we never use